### PR TITLE
Add  OAI-PMH endpoint 

### PR DESCRIPTION
**Pull request by little9**
This adds OAI-PMH https://www.openarchives.org/OAI/openarchivesprotocol.html functionality to 
the repository. 

It also adds request specs that test basic OAI functionality.

It adds the `blacklight_oai_provider` provider Gem and configures
it so that there is a single set that contains all records with a
title. The basic configuration in the gem will stay the same, but
some details (admin email address) will probably need to be changed
before going live.

### CODE REVIEW

File: app/controllers/catalog_controller.rb [hunk 1]
```diff
@@ -54,6 +55,25 @@ def enforce_show_permissions(_opts = {})
     config.view.slideshow.partials = [:index]
     config.view.slideshow.slideshow_method = :choose_image
 
+    config.oai = {
+      provider: {
+        repository_name: "Alexandria Digital Research Library",
+        repository_url: "https://alexandria.ucsb.edu/catalog/oai",
+        record_prefix: "oai:ucsb",
+        admin_email: "systems@curationexperts.com",
```
Comment (dunn) [inline app/controllers/catalog_controller.rb]: This should probably fetch from `Rails.application.secrets`.

**Comment by little9** [comment]
@dunn Could you take a look at this again? I put the server configuration in `config/secrets.yml.template`

**Comment by little9** [comment]
@bess I changed the title of the commit, I think the original was from a squash I did at some point

**Comment by dunn** [comment]
It won't matter usually, but hitting the endpoint errors when there are no objects in the repository:
```
NoMethodError (undefined method `timestamp' for nil:NilClass):

blacklight_oai_provider (6.0.0) lib/blacklight_oai_provider/solr_document_wrapper.rb:24:in `earliest'
oai (0.4.0) lib/oai/provider/response/identify.rb:16:in `block (2 levels) in to_xml'
builder (3.2.3) lib/builder/xmlbase.rb:175:in `_nested_structures'
builder (3.2.3) lib/builder/xmlbase.rb:68:in `tag!'
builder (3.2.3) lib/builder/xmlbase.rb:93:in `method_missing'
oai (0.4.0) lib/oai/provider/response/identify.rb:7:in `block in to_xml'
oai (0.4.0) lib/oai/provider/response.rb:40:in `block in response'
builder (3.2.3) lib/builder/xmlbase.rb:175:in `_nested_structures'
builder (3.2.3) lib/builder/xmlbase.rb:68:in `tag!'
oai (0.4.0) lib/oai/provider/response.rb:37:in `response'
oai (0.4.0) lib/oai/provider/response/identify.rb:6:in `to_xml'
oai (0.4.0) lib/oai/provider.rb:259:in `identify'
oai (0.4.0) lib/oai/provider.rb:309:in `process_request'
blacklight_oai_provider (6.0.0) app/controllers/concerns/blacklight_oai_provider/controller.rb:18:in `oai'
```

**Pull request merged**